server_src/routes/printers.js

```diff
@@ -125,25 +127,26 @@ router.get("/groups", ensureAuthenticated, async (req, res) => {
 });
 router.post("/printerInfo", ensureAuthenticated, async (req, res) => {
   const id = req.body.i;
+  const returnPrinter = await PrinterClean.returnPrintersInformation(id);
+  res.send(returnPrinter);
+});
 
-  const printers = await PrinterClean.returnPrintersInformation();
-  if (typeof id === "undefined" || id === null) {
-    res.send(printers);
-  } else {
-    const index = _.findIndex(printers, function (o) {
-      //Make sure ID's are both strings to stop recursion issues
-      return o._id == id;
-    });
-    const returnPrinter = {
-      printerName: printers[index].printerName,
-      apikey: printers[index].apikey,
-      _id: printers[index]._id,
-      printerURL: printers[index].printerURL,
-      storage: printers[index].storage,
-      fileList: printers[index].fileList,
-      systemChecks: printers[index].systemChecks
-    };
-    res.send(returnPrinter);
+router.post("/updatePrinterSettings", ensureAuthenticated, async (req, res) => {
+  const id = req.body.i;
+  if (!id) {
+    logger.error("Printer Settings: No ID key was provided");
+    res.statusMessage = "No ID key was provided";
```

Comment:
I dont think this statusMessage scales. Reason is that error response often is more than just a message. It could become a code, a context object, a message - more than just a string. So, I foresee a moment where we'd have to revert this in future and that's why it doesnt work for me.

`throw "message"`
`throw new Error("error message")`
`throw new ValidationError(validationErrorInput)`
`throw new OctoPrintError(octoprintParseError)`
... others

All of these dont fit in statusMessage string.

Reply:
Oh btw, that does not stop this PR. It is just that I would not like to do this globally. That's all im sayin

Reply:
I'm not using the status message for those eventualities there though? It's literally just saying "dodo, you need to provide a printer Id for this to work". Errors are caught below and didn't fire the status message? Can't remember. Anyway I think a global error handler should be tackled in another PR and update to use it. Would be good V2 prep work no?

Reply:
If your intention is to return input validation result, I propose to use `npm-input-validator` and return that result **in the body** + **400 code**. Much shorter code-wise, much clearer for another developer and much easier for the client to recognize and work with. This abstraction comes with so much benefit!
Here's how the input validation rules can look:
```
const checkInputRules = {
  persistenceUuid: "required",
  deviceUuid: "required",
  host: "string",
  port: "required|numeric",
  docker: "required|boolean",
  allowCrossOrigin: "required|boolean"
};
```
and this is the response if I leave out properties on purpose:
```
{
    "deviceUuid": {
        "message": "The device uuid field is mandatory.",
        "rule": "required"
    },
    "port": {
        "message": "The port field is mandatory.",
        "rule": "required"
    },
    "docker": {
        "message": "The docker field is mandatory.",
        "rule": "required"
    }
}
```
I had to do nothing for it (except for calling the validation and returning it if failed)! And because the rules are constant strings, we dont have to guess what is the reason for failure on the client side. Its explicit and reliable.

If input validation is what you're looking to abstract, I hope you like the method I propose above. This would make porting API code over to V2 an absolute breeze! 💨 💨💨